employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/AbstractPersistable.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.domain;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.Version;
+import javax.validation.constraints.NotNull;
+
+import org.optaplanner.core.api.domain.lookup.PlanningId;
+
+@MappedSuperclass
+public abstract class AbstractPersistable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @PlanningId
+    protected Long id;
```

Comment:
For the same of encapsulation, this one and another fields below could be declared as private and each descendent class would use getters to access them.